### PR TITLE
Framework: removed last remaining instance of SectionHeaderButton

### PR DESCRIPTION
**Pull request by rickybanister**
I did several things:

* Switched the 'Make Primary' domain button from the deprecated `SectionHeaderButton` to just a compact `Button`
* Deleted the component finally
* Updated the Devdocs
* Updated the Readme

**Before:**
screen shot 2015-12-10 at 10 50 21 am

**After:**
screen shot 2015-12-10 at 10 49 50 am

cc @ebinnion @mtias @breezyskies

### CODE REVIEW

File: client/components/section-header/docs/example.jsx [hunk 1]
```diff
@@ -23,16 +23,21 @@ var Cards = React.createClass( {
...
-					<SectionHeaderButton onClick={ function() {
-						alert( 'Clicked add button' );
-					} }>
-						Add
-					</SectionHeaderButton>
+					<Button
+						compact>
+						{ this.translate( 'Manage' ) }
+					</Button>
+					<Button
```
Comment (mtias) [inline client/components/section-header/docs/example.jsx]: Should we make one of these a `primary` one for the example?

**Comment by mtias** [comment]
Awesome, thanks for doing this.

**Comment by breezyskies** [comment]
`Make Primary` button looks good. :+1:

**Comment by rickybanister** [comment]
I fixed some css and updated the devdocs to show a primary action:

screen shot 2015-12-10 at 11 18 49 am

**Comment by rickybanister** [comment]
cc @shaunandrews as this affects #1391

File: client/components/section-header/docs/example.jsx [hunk 2]
```diff
@@ -23,16 +23,26 @@ var Cards = React.createClass( {
...
-					<SectionHeaderButton>Manage</SectionHeaderButton>
-					<SectionHeaderButton onClick={ function() {
-						alert( 'Clicked add button' );
-					} }>
-						Add
-					</SectionHeaderButton>
+					<Button
+						compact
+						primary>
+						{ this.translate( 'Primary Action' ) }
```
Comment (ebinnion) [inline client/components/section-header/docs/example.jsx]: I could be wrong, but I don't think we need to translate the strings in examples. If we do keep the translation, we should probably at least provide some context. Maybe something like,  `this.translate( 'Primary Action', { context: 'Button display in documentation' } )`?

cc @deBhal who might know whether we need to translate these strings.

__Note:__ There's also another translated string in this file on line `33`.

**Comment by ebinnion** [comment]
I'm not sure what route to use to test this component: `client/my-sites/upgrades/domain-management/edit/card/header/primary-domain-button.jsx`. I'm glad to test if you can help me out with that.

Code-wise, this LGTM, except comment about not using translated strings in example components, but I'm not sure which way that should go.

**Comment by rickybanister** [comment]
Please update the examples to fit the standard, I just saw translations in one file and not the other so I made it consistent. The contexts probably make sense too.

To test the domain button you have to visit a site that has a custom domain, then click on the domains sidebar item, then click on the non primary domain (i.e. example.wordpress.com) 

The url would be something like:

`https://wpcalypso.wordpress.com/domains/manage/example.wordpress.com/edit/example.com`

**Comment by ebinnion** [comment]
I went ahead and removed the translated strings. I also was able to test that the domain management section header worked fine. :+1:

**Comment by deBhal** [comment]
Sorry for the late reply, I've been out of commission.

These strings won't get picked up or translated, so there's no concern over wasting translator's time, it's just a matter of communications with developers. (We scan for strings after we bundle up the code, so we skip docs and tests automagically.  You can always `make translate` and check `calypso-strings.js` in the top level directory if you're not sure).

So it's just a question of whether it's better example code or not.  I'd lean towards including the translate calls in general, unless you feel like they obscure the point of the example.

I feel like in this example, you gain a little, particularly with the label attribute where you get a bit of extra signalling that this is a user-facing string and not a control value.

I've opened PR #1536 because I feel bad about being away and missing the other PR, not to be pushy.  If you feel like the example stands better without the changes, feel free to comment to that effect and close it.

**Pull request merged**